### DISCUSSION TITLE
Question about layer coloring

### DISCUSSION THREAD

**Discussion by AlexxCrp** [DomCR/ACadSharp#220] [Q&A] [answered]
I want to create a layer with a color and then add entities in the doc on that layer and for them to have the color of the Layer.

Color color = new Color(1);
Layer layer = new Layer("LayerTest")
{
    Color = color,
};
doc.Layers.Add(layer);

From my understanding if I do this and then start adding entities, like a line with its layer specified as "LayerTest", the line should have the Layer's color.

When I Do everything from above and then open the new Document, I see that the layer has been created but its color is still white (instead of red in this particular example). Am I doing something wrong or the layer coloring isn't working yet?

**Comment by DomCR** [comment, marked as answer]
Hi @AlexxCrp,

Yesterday I merged a PR to fix this issue https://github.com/DomCR/ACadSharp/pull/219, for this particular case when you add entities make sure that you are assigning the layer to the entity and the color is set to `ByLayer` (is the default but make sure is set as it is). Here is the example that I've used to test this feature:

```C#
Layer layer = new Layer("Test");
layer.Color = new Color(25);
this.Document.Layers.Add(layer);

Circle c = new Circle();
c.Center = new XYZ(0, 0, 0);
c.Radius = 10;
c.Layer = layer;
c.Color = Color.ByLayer;

this.Document.Entities.Add(c);
```

If you are using the nuget package, let me know and I'll do a release or if you are using the repository make sure to update the master branch with the latest changes.

Hope this helps.

**Reply by AlexxCrp** [reply]
Thank you so much for the quick reply, I am using the NuGet
